src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -26,538 +29,576 @@
-    session.script_to_generate = 'myScript0'
-    session.commit()
-    session.delete_script('myScript0')
-    actual_sample_rate = session.arb_sample_rate
-    in_range = abs(actual_sample_rate - 10000000) <= max(1e-09 * max(abs(actual_sample_rate), abs(10000000)), 0.0)   # https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python
-    assert in_range is True
-
-
-def test_reset(session):
-    default_output_mode = session.output_mode
-    assert default_output_mode == nifgen.OutputMode.ARB
-    session.output_mode = nifgen.OutputMode.SEQ
-    assert session.output_mode == nifgen.OutputMode.SEQ
-    session.reset()
-    assert session.output_mode == nifgen.OutputMode.ARB
-
-
-def test_reset_device(session):
-    default_trigger_mode = session.trigger_mode
-    assert default_trigger_mode == nifgen.TriggerMode.CONTINUOUS
-    session.trigger_mode = nifgen.TriggerMode.STEPPED
-    non_default_trigger_mode = nifgen.TriggerMode.STEPPED
-    assert non_default_trigger_mode == nifgen.TriggerMode.STEPPED
-    session.reset_device()
-    assert session.trigger_mode == nifgen.TriggerMode.CONTINUOUS
-
-
-def test_reset_with_default(session):
-    default_sample_rate = session.arb_sample_rate
-    assert default_sample_rate == 250000000.0
-    session.arb_sample_rate = 100000000.0
-    non_default_arb_sample_rate = session.arb_sample_rate
-    assert non_default_arb_sample_rate == 100000000.0
-    session.reset_with_defaults()
-    assert session.arb_sample_rate == 250000000.0
-
-
-def test_write_waveform_from_list(session):
-    data = [0.1] * 10000
-    session.write_waveform(session.allocate_waveform(len(data)), data)
-
-
-def test_write_waveform_from_numpy_array_float64(session):
-    data = numpy.ndarray(10000, dtype=numpy.float64)
-    data.fill(0.5)
-    session.write_waveform(session.allocate_waveform(len(data)), data)
-
-
-def test_write_waveform_numpy_array_int16(session):
-    data = numpy.ndarray(10000, dtype=numpy.int16)
-    data.fill(256)
-    session.write_waveform(session.allocate_waveform(len(data)), data)
-
-
-def test_write_named_waveform_from_list(session):
-    data = [0.1] * 10000
-    session.allocate_named_waveform('foo', len(data))
-    session.write_waveform('foo', data)
-
-
-def test_write_named_waveform_from_numpy_array_float64(session):
-    data = numpy.ndarray(10000, dtype=numpy.float64)
-    data.fill(0.5)
-    session.allocate_named_waveform('foo', len(data))
-    session.write_waveform('foo', data)
-
-
-def test_write_named_waveform_numpy_array_int16(session):
-    data = numpy.ndarray(10000, dtype=numpy.int16)
-    data.fill(256)
-    session.allocate_named_waveform('foo', len(data))
-    session.write_waveform('foo', data)
-
-
-def test_write_waveform_wrong_type(session):
-    waveform_handle = session.allocate_waveform(100)
-    for data in invalid_waveforms:
+    # Test doesn't run over gRPC because the exception isn't caught from create_advanced_arb_sequence().
```

Comment:
Not sure what exception you're talking about, but it sounds like this is something we need to fix.

Reply:
I've looked into this, and it actually seems like the original test might not be working correctly even when run locally. I created a bug (internal work item 2207462) to look into this and fix it in a separate change.